packages/bootleg-app-contracts/test/BootlegTraderApp.test.js

```diff
@@ -0,0 +1,120 @@
+const BootlegTraderApp = artifacts.require('BootlegTraderApp');
+const BootlegToken = artifacts.require('BootlegToken');
+
+contract('BootlegTraderApp', accounts => {
+  let tokenContract;
+  let app;
+  let result;
+  const tokenId = web3.utils.asciiToHex('super awesome token id');
+  const mockURI = 'https://ipfs.infura.io/ipfs/QmSomeHash';
+  const firstOwner = accounts[0];
+  const secondOwner = accounts[1];
+  const thirdOwner = accounts[2];
+  const notAnOwner = accounts[9];
+  const oneEthInWei = web3.utils.toWei('1', 'ether');
+  const twoEthInWei = web3.utils.toWei('2', 'ether');
+
+  beforeEach(async () => {
+    tokenContract = await BootlegToken.new('Bootleg', 'BLEG', 5);
+    app = await BootlegTraderApp.new(tokenContract.address);
+    await tokenContract.mintWithTokenURI(firstOwner, tokenId, mockURI);
+    await tokenContract.approve(app.address, tokenId);
+    await tokenContract.setAp;
+  });
+
+  describe('with a zero price', async () => {
+    it('has a price of zero by default', async () => {
+      expect((await app.getTokenPrice(tokenId)).toString()).toEqual('0');
+    });
+
+    it('cannot be purchased with a zero price', async () => {
+      let errorMsg = '';
+
+      try {
+        await app.purchase(tokenId, secondOwner, {
+          value: oneEthInWei
+        });
+      } catch (e) {
+        errorMsg = e.toString();
```

Comment:
I've always used `e.message` 🤷‍♂️ Didn't know this existed.

Reply:
This is what I found on the internets about it. I could use `e.message` I like that better